DuckDuckGo/EmailWaitlist.swift

```diff
@@ -46,6 +46,19 @@ private class EmailWaitlistRequestDelegate: EmailManagerRequestDelegate {
         }
     }
     // swiftlint:enable function_parameter_count
+    
+    func emailManagerKeychainAccessFailed(accessType: EmailKeychainAccessType, error: EmailKeychainAccessError) {
+        var parameters = [
+            PixelParameters.emailKeychainAccessType: accessType.rawValue,
+            PixelParameters.emailKeychainError: error.errorDescription
+        ]
+        
+        if case let .keychainAccessFailure(status) = error {
+            parameters[PixelParameters.emailKeychainKeychainStatus] = String(status)
+        }
+        
+        Pixel.fire(pixel: .emailAutofillKeychainError, withAdditionalParameters: parameters)
+    }
```

Comment:
I know that this code is duplicated a few times – all but one of these duplicated blocks is going away once #1190 is merged.